integrations/access/jira/types.go

```diff
@@ -18,11 +18,79 @@
 
 package jira
 
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
 // Jira REST API resources
 
+// ErrorResult is used to parse the errors from Jira.
+// The JSON Schema is specified here:
+// https://docs.atlassian.com/software/jira/docs/api/REST/1000.1223.0/#error-responses
+// However JIRA does not consistently respect the schema (especially for old instances).
+// We need to support legacy errors as well (array of strings).
 type ErrorResult struct {
-	ErrorMessages []string `url:"errorMessages"`
-	Errors        []string `url:"errors"`
+	ErrorMessages []string     `url:"errorMessages"`
+	Details       ErrorDetails `url:"errors"`
+}
+
+// Error implements the error interface and returns a string describing the
+// error returned by Jira.
+func (e ErrorResult) Error() string {
+	sb := strings.Builder{}
+	if len(e.ErrorMessages) > 0 {
+		sb.WriteString(fmt.Sprintf("error messages: %s ", e.ErrorMessages))
+	}
+	if details := e.Details.String(); details != "" {
+		sb.WriteString(fmt.Sprintf("error details: %s", details))
+	}
+	result := sb.String()
+	if result == "" {
+		return "Unknown jira error"
+	}
+	return result
+}
+
+// ErrorDetails are used to unmarshall inconsistently formatted Jira errors
+// details.
+type ErrorDetails struct {
+	Errors       map[string]string
+	LegacyErrors []string
```

Comment:
Nit: Godocs here would be helpful explaining how "legacy errors" differ from "errors" and why would Jira return them differently.